### ISSUE TITLE
Harvest Proposal Workflow

### ISSUE THREAD

**Issue by narock** [ec-geolink/glharvest#3]
The current harvest proposal says that the harvest service will update the central triple store when new data dumps are available. Originally, the workflow was to have the harvested data pass through the Wright State team for inferencing and co-reference resolution and then on to Marymount where the central triple store was housed. I didn't see those steps explicitly spelled out in the current proposal. Are they implied or did the workflow and location of the central triple change?

**Comment by mbjones** [comment]
No, it didn't change, I think its just ambiguous language, and we should be clearer.  The plan is still for @amoeba to build an automated harvester that grabs the dumps when they are updated, merge them into the named graph for that data set, and then notify the Wright State infrastructure that new data are available.  Wright State tools will then process it, and merge it into the central store that you maintain (although I am unclear on how that will happen in an automated way).  I think the goal should be that repositories can push and update to their data, and within seconds or minutes that data will be reflected in our central graphs.

**Comment by amoeba** [comment]
Yep, @mbjones is right. Sorry for the ambiguous language. I'll update the proposal today and ping you in this issue.

**Comment by narock** [comment]
Thanks for the clarification. I agree, we need to sort out how the Wright State operations will work. I am not sure how long that process will take to run and how we'll pick up/ingest the results. Maybe we need to do this in two stages? Data provider updates go from your harvester service to Wright State as well as directly into the production triple store. That way they are immediately available. We can then try to update with the enhanced graphs from Wright State at some set interval. This probably needs to be sorted on a telecon. I'm not sure if I can make the call on the 18th. I thought I a saw that @mbjones will be away as well. Should we try to make this an agenda item for the following call on 12/2?

**Comment by amoeba** [comment]
That sounds like a good idea. It certainly sounds like we need to talk about this and I think having more people on the call to talk would be good.

**Comment by mbjones** [comment]
The call on 12/2 sounds great -- I'll be there for that.  However, in the
interest of progress, it would be good if we continue discussing via github
where possible.  In this case, I don't see the advantage of directly
copying the graphs from harvest.geolink.org to data.geolink.org, as those
would be direct copies of one another.  It's the Wright State processing
that adds the pattern magic before it gets inserted into data.geolink.org.

**Comment by narock** [comment]
Will harvest.geolink.org have a publicly available SPARQL endpoint? If so, then I agree. Copying them to data.geolink.org would be redundant. I didn't realize you might serve the graphs to the community. I thought the harvester just harvested and data.geolink.org was handling the requirement to have graphs available within minutes of data providers releasing them. Works for me to have harvest.geolink.org harvest and make immediately available. Then WSU will process behind the scenes and at some to be determined intervals data.geolink.org will be updated with enhanced graphs.

**Comment by mbjones** [comment]
We hadn't discussed whether the harvest SPARQL endpoint would be public -- it would definitely be available to Wright State, as they said they would likely do their processing on the same node, and to you as well.  I have no problem with it being public either, whatever people want.

**Comment by amoeba** [comment]
I've updated the proposal document to match with my latest understanding of what we're doing. It could use a look and I'd appreciate feedback.
